packages/app-mobile/components/NoteEditor/ImageEditor/ImageEditor.tsx

```diff
@@ -0,0 +1,215 @@
+import ExtendedWebView from '../../ExtendedWebView';
+import { clearAutosave, writeAutosave } from './autosave';
+
+const logger = Logger.create('ImageEditor');
+
+type OnSaveCallback = (svgData: string)=> void;
+type OnCancelCallback = ()=> void;
+
+interface Props {
+	themeId: number;
+	initialSVGData: string;
+	onSave: OnSaveCallback;
+	onCancel: OnCancelCallback;
+}
+
+const useCss = (editorTheme: Theme) => {
+	return useMemo(() => {
+		return `
+			:root .imageEditorContainer {
+				--primary-background-color: ${editorTheme.backgroundColor};
+				--primary-background-color-transparent: ${editorTheme.backgroundColorTransparent};
+				--secondary-background-color: ${editorTheme.selectedColor2};
+				--primary-foreground-color: ${editorTheme.color};
+				--secondary-foreground-color: ${editorTheme.color2};
+				--primary-shadow-color: ${editorTheme.colorFaded};
+
+				width: 100vw;
+				height: 100vh;
+				box-sizing: border-box;
+			}
+
+			body, html {
+				padding: 0;
+				margin: 0;
+			}
+		`;
+	}, [editorTheme]);
+};
+
+const ImageEditor = (props: Props) => {
+	const editorTheme: Theme = themeStyle(props.themeId);
+	const webviewRef = useRef(null);
+
+	const onRequestCloseEditor = useCallback(() => {
+		Alert.alert(
+			_('Save changes?'), _('This drawing may have unsaved changes.'), [
+				{
+					text: _('Discard changes'),
+					onPress: async () => {
+						await clearAutosave();
+						props.onCancel();
+					},
+					style: 'destructive',
+				},
+				{
+					text: _('Save changes'),
+					onPress: () => {
+						// saveDrawing calls props.onSave(...) which may close the
+						// editor.
+						webviewRef.current.injectJS('saveDrawing();');
+					},
+				},
+			]
+		);
+		return true;
+	}, [webviewRef, props.onCancel]);
+
+	useEffect(() => {
+		BackHandler.addEventListener('hardwareBackPress', onRequestCloseEditor);
+
+		return () => {
+			BackHandler.removeEventListener('hardwareBackPress', onRequestCloseEditor);
+		};
+	}, [onRequestCloseEditor]);
+
+	const css = useCss(editorTheme);
+	const html = useMemo(() => `
+		<!DOCTYPE html>
+		<html>
+			<head>
+				<meta charset="utf-8"/>
+				<meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=no"/>
+
+				<style>
+					${css}
+				</style>
+			</head>
+			<body></body>
+		</html>
+	`, [css]);
+
+	const injectedJavaScript = useMemo(() => `
+		window.onerror = (message, source, lineno) => {
+			window.ReactNativeWebView.postMessage(
+				"error: " + message + " in file://" + source + ", line " + lineno
+			);
+		};
+
+		const saveDrawing = (isAutosave) => {
+			const img = window.editor.toSVG();
+			window.ReactNativeWebView.postMessage(
+				JSON.stringify({
+					action: isAutosave ? 'autosave' : 'save',
+					data: img.outerHTML,
+				}),
+			);
+		};
+		window.saveDrawing = saveDrawing;
+
+		const closeDrawing = () => {
+			window.ReactNativeWebView.postMessage('{ "action": "close" }');
```

Comment:
When there are no changes to the drawing (the undo stack is empty), a different action should be fired, one that doesn't show a "are you sure you want to close" dialog.